frontend/src/pages/Metrics.jsx

```diff
@@ -0,0 +1,118 @@
+import React, { useEffect, useState } from 'react'
+import Layout from '../layouts/Layout.jsx'
+import { NavLink, useParams } from 'react-router-dom'
+import '../styles/pages/Metrics.css'
+import '@fortawesome/fontawesome-free/css/all.min.css';
+
+const MetricRaws = (props) => {
+    useEffect(() => {
+        const fetchMetrics = async () => {
```

Comment:
if you use the getAllMetrics and setMetricData and setLoading from the parent component, then let the parent component calls them and pass the results to its child!